File: src/main/java/org/noxet/noxetserver/UsernameStorageManager.java
package org.noxet.noxetserver;

import org.bukkit.OfflinePlayer;

import java.util.UUID;

public class UsernameStorageManager extends ConfigManager {
    public UUID getUUIDFromUsernameOrUUID(String usernameOrUUID) {
        String uuid = config.getString(usernameOrUUID.toLowerCase());

        if(uuid == null) {
            // Try parsing it as a UUID directly instead.
            try {
                return UUID.fromString(usernameOrUUID);
            } catch(IllegalArgumentException e) {
                return null;
            }
        }

        return UUID.fromString(uuid);
    }

    public static String getCasedUsernameFromUUID(UUID uuid) {
        if(uuid == null)
            return null;

        OfflinePlayer offlinePlayer = NoxetServer.getPlugin().getServer().getOfflinePlayer(uuid);

        return offlinePlayer.getName(); // Username with casing if found, otherwise null.
    }

    public void assignUUIDToUsername(String username, UUID uuid) {
        config.set(username.toLowerCase(), uuid.toString());
        save();
    }

    @Override
    protected String getFileName() {
        return "usernames";
    }
}

File: src/main/java/org/noxet/noxetserver/util/UsernameStorageManager.scala
package org.noxet.noxetserver.util

import org.noxet.noxetserver.NoxetServer

import java.util.UUID
import scala.util.Try

object UsernameStorageManager extends ConfigManager:
  protected override def getFileName: String = "usernames"

  private def getUuidFromString(uuid: String): Option[UUID] = Try(UUID.fromString(uuid)).toOption

  def getUuidFromUsernameOrUuid(usernameOrUuid: String): Option[UUID] =
    getUuidFromString(
      Option(config.getString(usernameOrUuid.toLowerCase)).getOrElse(usernameOrUuid))

  def bindUsernameToUuid(username: String, uuid: UUID): Unit =
    config.set(username.toLowerCase, uuid.toString)
    save()

  def getCasedUsernameFromUuid(uuid: UUID): Option[String] =
    Option(NoxetServer.getPlugin.getServer.getOfflinePlayer(uuid).getName)